### PR TITLE
[Windows] CUDA installation fixes, add 11.1 and document absence of nvcuda.dll

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

I have been trying to provide `nvcuda.dll` in https://github.com/conda-forge/openmm-feedstock/pull/36 only to realize it is not possible, so I am documenting it here in case somebody wants to try again.

- The CUDA toolkit installer is supposed to provide the drivers too, but it will silently fail to do that if no GPU is present
- I thought that maybe getting the Tesla/datacenter drivers would allow a GPU-less install. Well, it's not the case. 
- Old versions of the drivers allowed to manually extract `nvcuda64.dl_` into `C:\Windows\system32` as `nvcuda.dll`, but that doesn't seem to be the case with most recent ones -- DLL will fail to load with error 193.

Basically the only way to provide `nvcuda.dll` involves breaking the EULA since you would have to redistribute the DLL from a working installation to bypass the hardware checks, and we can't do that.

Since I had to experiment with a few things to arrive to this conclusion, I am also adding a few fixes:

- Enable path to CUDA 11.1 (coming soon)
- Make sure we wait for the installers to finish!
- Install only the needed components (no samples, documentation, etc) to save time and space
- Make sure the CUDA failures are reflected as such in the CI status

## Status

- [X] (Linux only) Blocked by https://github.com/conda-forge/docker-images/pull/161 & https://github.com/conda-forge/docker-images/pull/163
- [X] (All platforms) `cudatoolkit` 11.1 is only available in `conda-forge/label/test` for now. 
- [x] Trying to fix https://github.com/conda-forge/nvcc-feedstock/issues/53

### CODE REVIEW

**Comment by jaimergp** [comment]
@conda-forge/core we need this for CUDA 11.1, now [merged](https://github.com/conda-forge/cudatoolkit-feedstock/pull/39). How should I proceed here? Merge `master`? Rebase? New PR?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% set name = "conda-forge-ci-setup" %}
-{% set version = "3.5.0" %}
+{% set version = "3.5.1" %}
```
Comment (beckermr) [inline recipe/meta.yaml:2]: Make sure this version is above the one on master.

**Comment by beckermr** [comment]
Resolve the conflicts and make sure the version is ahead. Then we should be good to go.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

Just to make sure the merge went ok.

**Comment by beckermr** [comment]
Note this won't build for travis. Also, we might want some code review from someone more knowledgeable than me on the CI setup package itself. cc @conda-forge/core

**Comment by jaimergp** [comment]
I also forgot I need to add the CUDA 11.1 matrix entries, which will fail on Linux due to the missing Docker image (see https://github.com/conda-forge/docker-images/pull/160).

**Comment by beckermr** [comment]
Can you fork to a non-org place? That way others can rerender.

**Comment by jaimergp** [comment]
> Can you fork to a non-org place? That way others can rerender.

Oops, I didn't know that was a limitation. I'll transfer back to my personal account.

**Comment by jakirkham** [comment]
It is. Though one can also re-render locally.

**Comment by jaimergp** [comment]
@conda-forge/core this is ready for review!

**Review by kkraus14 — APPROVED** [review]

**Comment by jaimergp** [comment]
May I ask for a merge if nobody is against it? 😄

**Pull request merged**